lib/errors.js

```diff
@@ -0,0 +1,61 @@
+class MapeoError extends Error {
+  constructor (code, message, innerError) {
+    super(message)
+    this.code = code
+    this.message = message
+    this.err = innerError
+  }
+}
+
+class PeerNotFoundError extends MapeoError {
+  constructor (innerError) {
+    super('ERR_PEER_NOT_FOUND', 'peer was not found', innerError)
+  }
+}
+
+class IncompatibleProjectsError extends MapeoError {
+  constructor (innerError) {
+    super('ERR_DIFF_PROJECT_KEYS', 'remote project key is different than ours', innerError)
+  }
+}
+
+class PrematureSyncError extends MapeoError {
+  constructor (innerError) {
+    super('ERR_PREMATURE_SYNC',  'sync started before handshake finished', innerError)
+  }
+}
+
+class UnsupportedSyncfileError extends MapeoError {
+  constructor (format, innerError) {
+    super('ERR_UNSUPPORTED_SYNCFILE_FORMAT', 'syncfile type is not supported ('+format+')', innerError)
+  }
+}
+
+class ConnectionLostError extends MapeoError {
+  constructor (innerError) {
+    super('ERR_CONNECTION_LOST', 'peer network connection lost', innerError)
+  }
+}
+
+class SyncError extends MapeoError {
+  constructor (message, innerError) {
+    super('ERR_SYNC', message || 'sync failed', innerError)
+  }
+}
+
+class MalformedProjectKeyError extends MapeoError {
+  constructor (key, innerError) {
+    super('ERR_MALFORMED_PROJECT_KEY', 'project key is malformed', innerError)
+    this.key = key
```

Comment:
Just in case this turns up in logs (e.g. if we are using a log service like bugsnag which logs the whole Error object, not just `Error.message`), I think this should only be the first 5 characters or something of the key, to avoid accidentally leaking it in the future.

Reply:
I think it should be ok: this error is only thrown when the key is malformed anyways.

Reply:
I changed my mind -- I agree. :)